### PR TITLE
KOGITO-243 - Align dependencies with Quarkus to avoid enforcer issues

### PR DESCRIPTION
**Pull request by mswiderski**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -65,14 +65,14 @@
     <version.antapacheregexp>1.8.2</version.antapacheregexp>
     <version.org.reflections>0.9.11</version.org.reflections>
     <version.javax.inject>1</version.javax.inject>
-    
-    <version.org.infinispan>10.0.0.Beta3</version.org.infinispan>
-    <version.org.infinispan.protostream>4.3.0.Alpha6</version.org.infinispan.protostream>
+
+    <version.org.infinispan>10.0.0.Beta5</version.org.infinispan>
```
Comment (ricardozanini) [inline pom.xml]: Why not [10.0.0.CR1](https://mvnrepository.com/artifact/org.infinispan/infinispan-core/10.0.0.CR1)? This is the version I've been using with the operator tests.

**Review by ricardozanini — CHANGES_REQUESTED** [review]
Just a small note regarding the Infinispan version. Quarkus alignment?

File: pom.xml (hunk 1 above)
Comment (cristianonicolai) [reply pom.xml]: I guess the idea is just to align with whatever Quarkus is using

**Review by cristianonicolai — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (mswiderski) [reply pom.xml]: that's correct, it is to align with what quarkus currently define.

**Comment by mswiderski** [comment]
@cristianonicolai @ricardozanini I had to downgrade Infinispan as it does not work due to changes in the code used by our tests - there is interface change that breaks as we have mixed dependencies due to quarkus bom bringing in 10.0.0-Beta3 and our code bringing Beta5 and thus it mixes it and fails with NoSuchMethodException

**Comment by ricardozanini** [comment]
At least the CR1 server is working fine with Beta3 client.

**Comment by mswiderski** [comment]
it does work for runtime but it fails for data index service tests that get Infinispan libs mixed up - some from the Beta3 and some from Beta5

**Pull request merged**